app/domain/mail_relay/lists.rb

```diff
@@ -124,10 +124,18 @@ def prepare_bounced_message(list_address, sender_address)
       message.smtp_envelope_from = env_sender
     end
 
+    def domain_is_valid?(domain)
+        domain !~ /^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$/
+    end
+
     def sender_is_additional_sender?
       additional_senders = mailing_list.additional_sender.to_s
       list = additional_senders.split(/[,;]/).collect(&:strip).select(&:present?)
-      list.include?(sender_email)
+      sender_domain = sender_email.sub(/^[^@]*@/, "*@")
```

Comment:
Prefer single quotes when you don't need string interpolation or special symbols.
```suggestion
      sender_domain = sender_email.sub(/^[^@]*@/, '*@')
```